### PR TITLE
ROC-6749 Add filter to include pragma and Cache-Control headers to all responses

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-6749

### Change description ###
Use a filter to add the `pragma: no-cache` and `Cache-Control: no-store` headers to all responses

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        filter = new NoCacheFilter();
+    }
+
+    @Test
+    void shouldAddHeaders() throws ServletException, IOException {
+        filter.doFilterInternal(request, response, chain);
+        assertAll(
+            () -> verify(response).addHeader("pragma", "no-cache"),
+            () -> verify(response).addHeader("Cache-Control", "no-store"),
+            () -> verifyZeroInteractions(request)
```
Comment (KiranVarma02) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java]: looks like `verifyZeroInteractions` is deprecated. `verifyNoMoreInteractions` can be used which has same behaviour.

Comment (andrewwa-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java]: Thanks, that didn't show up in my IDE for some reason. I'm changing it to `verifyNoInteractions()` which I feel communicates the sentiment better

File: src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@Component
+public class NoCacheFilter extends OncePerRequestFilter {
+    @Override
+    protected void doFilterInternal(
+        HttpServletRequest request,
+        HttpServletResponse response,
+        FilterChain filterChain
+    ) throws ServletException, IOException {
+        response.addHeader("Cache-Control", "no-store");
+        response.addHeader("pragma", "no-cache");
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java]: `Pragma` with P capital please

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4853020&bid=16100835)

**Pull request merged**